src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,77 @@
+package frc.robot.subsystems.intake;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+
+import static frc.robot.Constants.Intake.IS_COMPENSATING_VOLTAGE;
+
+public class Intake extends SubsystemBase {
+    private static Intake INSTANCE;
+    private final WPI_TalonSRX motor = new WPI_TalonSRX(Ports.Intake.MOTOR);
+    private final Solenoid retractor = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Intake.SOLENOID);
+
+    private Intake() {
+        motor.setInverted(Ports.Intake.IS_MOTOR_INVERTED);
+        motor.enableVoltageCompensation(IS_COMPENSATING_VOLTAGE);
+        motor.configVoltageCompSaturation(Constants.NOMINAL_VOLTAGE);
+    }
+
+
+    /**
+     * creates an instance of the intake.
+     *
+     * @return the subsystem instance.
+     */
+    public static Intake getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Intake();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Sets the power of the intake [%].
+     *
+     * @param power desired power in percentage.
+     */
+    public void setPower(double power) {
+        motor.set(ControlMode.PercentOutput, power);
+    }
+
+    /**
+     * Open intake retractor.
+     */
+    public void openRetractor() {
+        retractor.set(SolenoidState.OPEN.value);
+    }
+
+    /**
+     * Close intake retractor.
+     */
+    public void closeRetractor() {
+        retractor.set(SolenoidState.CLOSED.value);
+    }
+
+    /**
+     * Toggles intake retractor.
+     */
+    public void toggleRetractor() {
+        retractor.toggle();
+    }
+
+    public enum SolenoidState {

```

Comment:
Rename to `RetractrorState`, `IntakeState` or simply `State` (or another name you find fitting). First, to match `retractor`'s name, and second, to avoid exposing hardware details outside this class.